### PR TITLE
gh-94360: Fix a tokenizer crash when reading encoded files with syntax errors from stdin

### LINKED ISSUE #94360: Tokenizer crash when redirecting input to stdin
Hi!
We were doing some fuzzing using AFL for latest version 3.10.5, and we found an interesting issue.
There is some crash that may be exploitable. I check latest version from git and crash was reproduced.
So input file attached with screenshots

[input.tar.gz](https://github.com/python/cpython/files/8997102/input.tar.gz)

### PR DESCRIPTION
**Pull request by pablogsal**
* Issue: gh-94360

Closes: #94360

### CODE REVIEW

**Comment by pablogsal** [comment]
I'm not adding a test because this doesn't reproduce nicely on the test suite.

**Comment by zware** [comment]
Confirmed that this resolves the crash as produced by my reduced reproducer in `main` and  `3.11`, and without the change to `Parser/pegen_errors.c` (which doesn't exist in 3.10) it also resolves both reproducers for `3.10`.

File: Parser/pegen_errors.c [hunk 1]
```diff
@@ -259,15 +259,15 @@ get_error_line_from_tokenizer_buffers(Parser *p, Py_ssize_t lineno)
     const char* buf_end = p->tok->fp_interactive ? p->tok->interactive_src_end : p->tok->inp;
 
     for (int i = 0; i < relative_lineno - 1; i++) {
-        char *new_line = strchr(cur_line, '\n') + 1;
```
Comment (ambv) [inline Parser/pegen_errors.c:262]: `NULL + 1` is beautifully strongly typed, huh? 😎

File: Parser/tokenizer.c [hunk 2]
```diff
@@ -308,6 +308,10 @@ tok_concatenate_interactive_new_line(struct tok_state *tok, const char *line) {
 
     Py_ssize_t current_size = tok->interactive_src_end - tok->interactive_src_start;
     Py_ssize_t line_size = strlen(line);
+    char last_char = line[line_size > 0 ? line_size - 1: line_size];
```
Comment (ambv) [inline Parser/tokenizer.c]:
```suggestion
    char last_char = line[line_size > 0 ? line_size - 1 : line_size];
```

**Pull request merged**